### PR TITLE
Add implementation on of sys/time.h

### PR DESCRIPTION
**Pull request by Coquinho**
This should add all needed functions/structures from sys/time.h on a Windows fashion.

### CODE REVIEW

**Comment by caiosignor** [comment]
looks good

File: src/lib/evil/unposix/sys/time.h [hunk 1]
```diff
@@ -1,26 +1,43 @@
-#ifndef TIME_H
-#define TIME_H
-
-#include "unimplemented.h"
+#ifndef SYS_TIME_H
+#define SYS_TIME_H
 
+#ifndef WIN32_LEAN_AND_MEAN
+# define WIN32_LEAN_AND_MEAN 1
```
Comment (caiosignor) [inline src/lib/evil/unposix/sys/time.h]: what is this ?

Comment (Coquinho) [reply src/lib/evil/unposix/sys/time.h]: Window.h automatically includes a lot of other headers, using WIN32_LEAN_AND_MEAN it only includes what is explicitly included. I think that, for safety, everywhere you include windows.h you should verify if WIN32_LEAN_AND_MEAN is set (Am I right @JPTIZ ? ).

Comment (JPTIZ) [reply src/lib/evil/unposix/sys/time.h]: > Window.h automatically includes a lot of other headers, using WIN32_LEAN_AND_MEAN it only includes what is explicitly included. I think that, for safety, everywhere you include windows.h you should verify if WIN32_LEAN_AND_MEAN is set (Am I right @JPTIZ ? ).

Mostly that. Not exactly only what is explicitly included, just prevents messing up trying to use two conflicting APIs at the same time (e.g. winsock and winsock2).

**Comment by Coquinho** [comment]
@JPTIZ, @caiosignor there were some changes, could you have a look again?

**Pull request merged**